### DISCUSSION TITLE
How to generate mysql zerofill on db pull

### DISCUSSION THREAD

**Discussion by renatogabrielbr** [prisma/prisma#13988] [Q&A] [answered]
I have created a field tinyint 2 unsigned zerofill but when i do make a prisma db pull, does not work, 
Unsigned alone works, zerofill removes unsigned from db pull.

Mysql 8

**Comment by TasinIshmam** [comment, marked as answer]
Hi @tidytrax 👋  

It would be much easier for us to debug the problem if you could provide more details. Could you kindly provide some example code or clear steps to reproduce the problem?

**Reply by renatogabrielbr** [reply]
Hello, yes I can

I use mysql 8, and i have 2 fields tinyint(2) Unsigned ZeroFill
So when I run prisma db pull, 
This changes does not show(prisma shema), 
If i run prisma db push or migrate does not reflect also.
If I just use unsigned works,
if I use zerofill, unsigned is not generate and zerofill in both cases does not generate with or without unsigned 

I use last prisma version
To fix it i had to put on first migration an Alter Table to do the changes after import the database if use migration, for db push I didn`t find any solution

can be also smallint, int...

```
CREATE TABLE `test`.`test`  (
  `field1` tinyint(2) UNSIGNED ZEROFILL NULL
)
```

`pnpm prisma db pull `

**Reply by TasinIshmam** [reply]
Thanks, I can replicate this now. 

Prisma does not support the `zerofill` attribute in `MySQL`. I think the workaround you figured out (manually modifying the migration file) is your best option for using this. 

We don't seem to have a feature request for this yet. If you're interested in seeing this get implemented, please feel free to open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and explain your use case!

**Reply by renatogabrielbr** [reply]
OK thank you , I will open soon, but the alter table on migration works fine,

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
